Update to embkafka 20

ttnesby opened the pull request:
Upgraded to embeddedkafka 2.x with security.
Also bumped the versions for other components and fixed deprecated cases
Finally, added the missing test cases (create, delete, acls for topics)
No feature enhancements

Afterwards, adminRest is ready for the other issues

ttnesby committed:
- Update to embedded kafka 2.0 with security
- Update to kafka 2.0
- Update of other components
- Added missing tests for topic (create, delete and ACL)

ttnesby committed:
- Finalized test for correct ACL for KP- and KC- groups

ttnesby committed:
- Updated embeddedkafka to latest version

tronghn commented on src/main/kotlin/no/nav/integrasjon/api/nais/client/NaisClient.kt:
```diff
@@ -47,14 +47,14 @@ fun Routing.getIsReady(adminClient: AdminClient, config: FasitProperties) =
                     Pair(HttpStatusCode.OK, "is ready")
                 else
-                    Pair(HttpStatusCode.ExceptionFailed, "is not ready")
+                    Pair(HttpStatusCode.ExpectationFailed, "is not ready")
```
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/417

> The HTTP 417 Expectation Failed client error response code indicates that the expectation given in the request's Expect header could not be met.

As far as I can see, we do not use or require any `Expect` headers.

The 4xx series of HTTP error codes indicates a client error - I think it'd be more semantically correct to send back a 500 Internal Server Error instead if a request results in an exception.

Same comment for every other usage of this HTTP status code.

tronghn replied:
Also, if the request failed due to an error in the request (e.g. missing or wrong parameters), a HTTP 400 should be returned.

ttnesby replied:
Concur :-)

ttnesby committed:
- Ongoing refactoring of response category (400 versus 500) error messages

ttnesby committed:
- More ongoing refactoring of response category (400 versus 500)
  error messages

ttnesby committed:
- Even more ongoing refactoring of response category (400 versus 500)
  error messages, fixed most of the existing tests. Oneshot left and
  one of the tests for update topic config

ttnesby committed:
- Finished with refactoring

tronghn committed:
Return correct HTTP status code
